js/plugins/views.js

```diff
@@ -0,0 +1,73 @@
+var clinicalTimelineViewHandler = function(options) {
+  //initiliase simple-advanced buttons
+  var timeline = options.timeline;
+  var overviewSVG = d3.select(".overview");
+  if (options.timelineBool.advancedView) {
+    $("#advBtn").prop("disabled", true); //disable repeated clicking of advButton
+    $("#advBtn").addClass("active");
+    $("#simpleBtn").prop("disabled", false); //enable simpleBtn
+    $("#simpleBtn").removeClass("active");
+    $("#export-button").prop("disabled", true);
+    $("#jsonInput").css("display", "block");
+    $("#tooltip-controller").css("visibility", "hidden");
+    $("#vertical-trim-toggle").css("visibility", "hidden");
+    if (location.href.indexOf("?view=simple") > -1) {  
+      window.history.replaceState('advancedView', '', location.href.replace("?view=simple", "?view=advanced"));
+    }
+    initAdvancedView() 
+  } else {
+    $("#advBtn").prop("disabled", false); //enable advBtn
+    $("#advBtn").removeClass("active");
+    $("#simpleBtn").prop("disabled", true); //disable repeated clicking of simpleBtn
+    $("#simpleBtn").addClass("active");
+    $("#export-button").prop("disabled", false);
+    $("#jsonInput").css("display", "none");
+    $("#tooltip-controller").css("visibility", "visible");
+    $("#vertical-trim-toggle").css("visibility", "visible");
```

Comment:
These are way too many html ids that could result in problems when incorporating in the main cbioportal page. It's very likely someone else used `jsonInput` or `advBtn` or `export-button`, so try to use other type of selectors maybe `$(divId +" " + "export-button")`. 

Another thing is that this is all initialization and removal code for different plugins. It would be more clean to have a `remove` function on each plugin that removes all the plugins related html in a clean manner. You could also have an `init` function on each plugin if necessary. This is why some sort of plugin prototype can be useful, it's clear to everybody that extends from that you can override the `init`, `run` and `remove` function if necessary. You could make some tests there to make sure the timeline object is the same before adding the plugin and after removing the plugin.